game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculator.java

```diff
@@ -103,11 +103,25 @@ public AggregateResults calculate(
         final MustFightBattle battle =
             new MustFightBattle(location2, attacker2, gameData, battleTracker);
         battle.setHeadless(true);
+        if (amphibious) {
+          attackingUnits.forEach(
+              unit -> {
+                unit.getProperty(Unit.UNLOADED_AMPHIBIOUS)
```

Comment:
I have this code duplicated several times.  If you have any suggestions on how to shrink it, I'd love to know.

Reply:
I suspect `attackingUnits` would need to become a datatype. In this case you could put this logic behind a method. I don't think I'd advise to do that just yet, the amount of data copying could actually become significant. BattleCalculator is regrettably a giant mess *AND* performance sensitive, a horrible combination and place to be..

Reply:
Another thought, the API of the 'setProperty' is wonky with the exception type. If there were type safe overloads of the method without a throws, then the repetition would not be much of a problem at all.